File: add_back_buttons.py
import re

path = r'C:\Users\aaron\Documents\words-of-plainness\src\chapters\06-embrace-the-savior.njk'
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

lines = content.split('\n')
output = []
current_path = None
current_movement = None
in_reflection = False

for line in lines:
    m = re.search(r'class="movement-section".*?data-path="(\w+)".*?data-movement="(\d+)"', line)
    if m:
        current_path = m.group(1)
        current_movement = int(m.group(2))
        in_reflection = False
    m2 = re.search(r'class="reflection-section".*?data-path="(\w+)"', line)
    if m2:
        current_path = m2.group(1)
        in_reflection = True
        current_movement = None

    output.append(line)

    if '<div class="movement-nav">' in line:
        if not in_reflection and current_movement and current_movement >= 2:
            prev = current_movement - 1
            output.append('            <button class="btn-back" onclick="goBackOnPath()">&larr; Back to Movement {}</button>'.format(prev))
        if in_reflection:
            output.append('            <button class="btn-back" onclick="goBackOnPath()">&larr; Back to Movement 5</button>')

result = '\n'.join(output)
with open(path, 'w', encoding='utf-8') as f:
    f.write(result)

count = result.count('btn-back')
print('Done. btn-back occurrences: {} (15 buttons + 2 CSS rules = 17)'.format(count))

File: add_back_buttons.js
const fs = require('fs');

const path = 'C:\\Users\\aaron\\Documents\\words-of-plainness\\src\\chapters\\06-embrace-the-savior.njk';
const content = fs.readFileSync(path, 'utf8');
const lines = content.split('\n');
const output = [];

let currentPath = null;
let currentMovement = null;
let inReflection = false;

for (const line of lines) {
  const movMatch = line.match(/class="movement-section".*?data-path="(\w+)".*?data-movement="(\d+)"/);
  if (movMatch) {
    currentPath = movMatch[1];
    currentMovement = parseInt(movMatch[2]);
    inReflection = false;
  }
  const refMatch = line.match(/class="reflection-section".*?data-path="(\w+)"/);
  if (refMatch) {
    currentPath = refMatch[1];
    inReflection = true;
    currentMovement = null;
  }

  output.push(line);

  if (line.includes('<div class="movement-nav">')) {
    if (!inReflection && currentMovement && currentMovement >= 2) {
      const prev = currentMovement - 1;
      output.push(`            <button class="btn-back" onclick="goBackOnPath()">&larr; Back to Movement ${prev}</button>`);
    }
    if (inReflection) {
      output.push('            <button class="btn-back" onclick="goBackOnPath()">&larr; Back to Movement 5</button>');
    }
  }
}

const result = output.join('\n');
fs.writeFileSync(path, result, 'utf8');

const count = (result.match(/btn-back/g) || []).length;
console.log(`Done. btn-back occurrences: ${count} (15 buttons + 2 CSS rules = 17)`);
